image/templates/helm/shared/templates/_scanner-v4_init.tpl.htpl

```diff
@@ -84,12 +84,14 @@
     {{- if kindIs "invalid" $._rox.scannerV4.indexer.serviceTLS.generate }}
       {{/* We need special handling here, because 'generate' will default to 'false' on upgrades. */}}
       {{/* And in case scanner V4 was not deployed earlier, we need to make sure that these resources are correctly initialized. */}}
-      {{- $lookupOut := dict -}}
```

Comment:
Let me explain what is going on here.

  The problem during Helm-linting the chart was that an attempt was made to generate certs for Scanner V4 services even though no CA certificate was present. How did we end up here? In a nutshell: By default we are only generating new certs if `IsInstall == true`. But, in order to enable a smooth Scanner V4 installation experience we had a special-case in the code, which switched on cert generation for Scanner V4 services if a resource lookup indicated that no TLS secrets for Scanner V4 services exist yet (think: _during an upgrade_).

This special handling did not apply to the CA certificate as well, hence that one was not created => it didn't exist.

And when an attempt was made to generate certs for Scanner V4 services were, due to the above special handling => BOOM.

There are probably multiple possible solutions to this. I have tried here to find a solution which does not break any other unit tests, looks like this change is working:

I am protecting the special handling I have described previously with the condition

```
      {{- if $.Release.IsUpgrade -}}
      ...
      {{- end -}}
```

This means that only in case `IsUpgrade == true`, which should be the case for upgrades ;-), this whole special logic kicks in.

For Helm linting we have the situation `IsUpgrade == IsInstall == false`, hence this code path is not used and no attempt is made to generate certificates for Scanner V4 services. :tada:

Reply:
What's the value of `IsUpgrade` when using `helm upgrade --install` in case the helm release does not exist yet?

Reply:
Just tested:
```
IsInstall: true
IsUpgrade: false
```